bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -970,4 +970,162 @@ mod tests {
         init::init();
         assert!(init::fips_mode().unwrap().is_enabled());
     }
+
+    /// Test that an application context can be set and retrieved.
+    #[test]
+    fn test_app_context_set_and_retrieve() {
+        struct TestApplicationContext {
+            test_value: u32,
+        }
+
+        let mut connection = connection::Connection::new_server();
+        connection.set_application_context(TestApplicationContext { test_value: 1142 });
+
+        let context = connection
+            .application_context::<TestApplicationContext>()
+            .unwrap();
+        assert_eq!(context.test_value, 1142);
+    }
+
+    /// Test that an application context can be modified.
+    #[test]
+    fn test_app_context_modify() {
+        struct TestApplicationContext {
+            test_value: u32,
+        }
+
+        let mut connection = connection::Connection::new_server();
+        connection.set_application_context(TestApplicationContext { test_value: 0 });
+
+        let context = connection
+            .application_context_mut::<TestApplicationContext>()
+            .unwrap();
+        context.test_value += 1;
+
+        let context = connection
+            .application_context::<TestApplicationContext>()
+            .unwrap();
+        assert_eq!(context.test_value, 1);
+    }
+
+    /// Test that an application context can be overridden.
+    #[test]
+    fn test_app_context_override() {
+        struct TestApplicationContext {
+            test_value: u32,
+        }
+
+        let mut connection = connection::Connection::new_server();
+        connection.set_application_context(TestApplicationContext { test_value: 1142 });
+
+        {
+            let context = connection
+                .application_context::<TestApplicationContext>()
+                .unwrap();
+            assert_eq!(context.test_value, 1142);
+        }
```

Comment:
Generally it's not necessary to separate things out to separate scopes in rust since
- variables are `const` by default
- you can "shadow" variable declaration

But in this context, I think it'd be a bit neater to just directly assert on the context result.
```suggestion
        assert_eq!(
            connection.application_context::<TestApplicationContext>(),
            Some(1142)
        );
```

Reply:
Yeah this much better thanks!